server/db/models/Gig.js

```diff
@@ -10,11 +12,60 @@ module.exports = function (sequelize, DataTypes) {
   }, {
     classMethods: {
 
-      getGigs: function (organizationId) {
+      // accepts object like :
+      // {
+      //   startDate: STRING,
+      //   endDate: STRING ,
+      //   includeStaff: BOOLEAN,
+      //   organizationId: STRING
+      // }
+
+      getGigs: function (info) {
+        var start = moment(info.startDate);
+        var end = moment(info.endDate);
+        var include = [];
+
+        var staffQuery = {
+          model: Gig.associations['Users'].target,
+          attributes: ['id', 'name', 'email'],
+          through: {
+            attributes: [['PositionId', 'id'], 'admin_accepted', 'worker_accepted'],
+            as: 'Position'
+          },
+        };
+
+        var positionQuery = {
+          model: Gig.associations['Positions'].target,
+          attributes: ['title'],
+          through: {
+            attributes: ['required', 'filled'],
+            as: 'statusInfo'
+          }
+        };
+
+        include.push(positionQuery);
+        info.includeStaff && include.push(staffQuery); // just for day view
+
         return Gig.findAll({
           where: {
-            organizationId: organizationId
-          }
+            organizationId: info.organizationId,
+            date: {
+              $between: [start.format('YYYY-MM-DD'), end.format('YYYY-MM-DD')]
+            },
+          },
+          attributes: [
```

Comment:
If the attribute option is left off, would the query return all of the attributes. If so, it seems like you'd only want to specify attributes in order to explicitly ignore certain fields - otherwise take it all. Maybe this is necessary to ensure the foreign keys are included?